src/posix/platform/settings.cpp

```diff
@@ -260,13 +251,14 @@ otError otPlatSettingsAdd(otInstance *aInstance, uint16_t aKey, const uint8_t *a
 
     if (size > 0)
     {
-        VerifyOrDie(0 == lseek(sSettingsFd, 0, SEEK_SET));
+        VerifyOrDie(0 == lseek(sSettingsFd, 0, SEEK_SET), OT_EXIT_ERROR_ERRNO);
         swapWrite(swapFd, static_cast<uint16_t>(size));
     }
 
     VerifyOrDie(write(swapFd, &aKey, sizeof(aKey)) == sizeof(aKey) &&
-                write(swapFd, &aValueLength, sizeof(aValueLength)) == sizeof(aValueLength) &&
-                write(swapFd, aValue, aValueLength) == aValueLength);
+                    write(swapFd, &aValueLength, sizeof(aValueLength)) == sizeof(aValueLength) &&
+                    write(swapFd, aValue, aValueLength) == aValueLength,
+                OT_EXIT_ERROR_ERRNO);
```

Comment:
`errno` is not guaranteed to be set. Better to handle write() == -1 and write() < sizeof(aKey) separately if we want to output an error information.

Reply:
I think it is better to use  `OT_EXIT_FAILURE` to replace ` OT_EXIT_ERROR_ERRNO`  keep the logic consistent with original code.